{"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"UBNetworkingTaskRecoveryResult"}],"languages":["swift"],"platforms":["iOS"]}]}],"schemaVersion":{"major":0,"minor":3,"patch":0},"sections":[],"variants":[{"paths":["\/documentation\/ubfoundation\/ubnetworkingtaskrecoveryresult"],"traits":[{"interfaceLanguage":"swift"}]}],"identifier":{"url":"doc:\/\/UBFoundation\/documentation\/UBFoundation\/UBNetworkingTaskRecoveryResult","interfaceLanguage":"swift"},"abstract":[{"type":"text","text":"The result of attempting a recovery after a network task fails"}],"kind":"symbol","metadata":{"fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"UBNetworkingTaskRecoveryResult"}],"title":"UBNetworkingTaskRecoveryResult","roleHeading":"Enumeration","role":"symbol","symbolKind":"enum","externalID":"s:12UBFoundation30UBNetworkingTaskRecoveryResultO","modules":[{"name":"UBFoundation"}],"navigatorTitle":[{"kind":"identifier","text":"UBNetworkingTaskRecoveryResult"}]},"hierarchy":{"paths":[["doc:\/\/UBFoundation\/documentation\/UBFoundation"]]},"topicSections":[{"title":"Enumeration Cases","identifiers":["doc:\/\/UBFoundation\/documentation\/UBFoundation\/UBNetworkingTaskRecoveryResult\/cannotRecover","doc:\/\/UBFoundation\/documentation\/UBFoundation\/UBNetworkingTaskRecoveryResult\/recovered(data:response:info:)","doc:\/\/UBFoundation\/documentation\/UBFoundation\/UBNetworkingTaskRecoveryResult\/recoveryOptions(options:)","doc:\/\/UBFoundation\/documentation\/UBFoundation\/UBNetworkingTaskRecoveryResult\/restartDataTask"]}],"references":{"doc://UBFoundation/documentation/UBFoundation/UBNetworkingTaskRecoveryResult/restartDataTask":{"role":"symbol","title":"UBNetworkingTaskRecoveryResult.restartDataTask","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"restartDataTask"}],"abstract":[{"type":"text","text":"The strategy has finished recovering and needs to restart the task"}],"identifier":"doc:\/\/UBFoundation\/documentation\/UBFoundation\/UBNetworkingTaskRecoveryResult\/restartDataTask","kind":"symbol","type":"topic","url":"\/documentation\/ubfoundation\/ubnetworkingtaskrecoveryresult\/restartdatatask"},"doc://UBFoundation/documentation/UBFoundation/UBNetworkingTaskRecoveryResult/recoveryOptions(options:)":{"role":"symbol","title":"UBNetworkingTaskRecoveryResult.recoveryOptions(options:)","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"recoveryOptions"},{"kind":"text","text":"("},{"kind":"externalParam","text":"options"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"UBNetworkTaskRecoveryOptions","preciseIdentifier":"s:12UBFoundation28UBNetworkTaskRecoveryOptionsV"},{"kind":"text","text":")"}],"abstract":[{"type":"text","text":"The strategy could offer different options of recovery"}],"identifier":"doc:\/\/UBFoundation\/documentation\/UBFoundation\/UBNetworkingTaskRecoveryResult\/recoveryOptions(options:)","kind":"symbol","type":"topic","url":"\/documentation\/ubfoundation\/ubnetworkingtaskrecoveryresult\/recoveryoptions(options:)"},"doc://UBFoundation/documentation/UBFoundation/UBNetworkingTaskRecoveryResult/cannotRecover":{"role":"symbol","title":"UBNetworkingTaskRecoveryResult.cannotRecover","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"cannotRecover"}],"abstract":[{"type":"text","text":"Cannot recover the failure"}],"identifier":"doc:\/\/UBFoundation\/documentation\/UBFoundation\/UBNetworkingTaskRecoveryResult\/cannotRecover","kind":"symbol","type":"topic","url":"\/documentation\/ubfoundation\/ubnetworkingtaskrecoveryresult\/cannotrecover"},"doc://UBFoundation/documentation/UBFoundation/UBNetworkingTaskRecoveryResult":{"role":"symbol","title":"UBNetworkingTaskRecoveryResult","fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"UBNetworkingTaskRecoveryResult"}],"abstract":[{"type":"text","text":"The result of attempting a recovery after a network task fails"}],"identifier":"doc:\/\/UBFoundation\/documentation\/UBFoundation\/UBNetworkingTaskRecoveryResult","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"UBNetworkingTaskRecoveryResult"}],"url":"\/documentation\/ubfoundation\/ubnetworkingtaskrecoveryresult"},"doc://UBFoundation/documentation/UBFoundation/UBNetworkingTaskRecoveryResult/recovered(data:response:info:)":{"role":"symbol","title":"UBNetworkingTaskRecoveryResult.recovered(data:response:info:)","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"recovered"},{"kind":"text","text":"("},{"kind":"externalParam","text":"data"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Data","preciseIdentifier":"s:10Foundation4DataV"},{"kind":"text","text":"?, "},{"kind":"externalParam","text":"response"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"HTTPURLResponse","preciseIdentifier":"c:objc(cs)NSHTTPURLResponse"},{"kind":"text","text":", "},{"kind":"externalParam","text":"info"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"UBNetworkingTaskInfo","preciseIdentifier":"s:12UBFoundation20UBNetworkingTaskInfoV"},{"kind":"text","text":"?)"}],"abstract":[{"type":"text","text":"The strategy has finished successfully and could recover the data"}],"identifier":"doc:\/\/UBFoundation\/documentation\/UBFoundation\/UBNetworkingTaskRecoveryResult\/recovered(data:response:info:)","kind":"symbol","type":"topic","url":"\/documentation\/ubfoundation\/ubnetworkingtaskrecoveryresult\/recovered(data:response:info:)"},"doc://UBFoundation/documentation/UBFoundation":{"role":"collection","title":"UBFoundation","abstract":[],"identifier":"doc:\/\/UBFoundation\/documentation\/UBFoundation","kind":"symbol","type":"topic","url":"\/documentation\/ubfoundation"}}}